MaterialLib/MPL/CreateComponent.cpp

```diff
@@ -72,28 +72,29 @@ std::vector<std::unique_ptr<Component>> createComponents(
     }
 
     std::vector<std::unique_ptr<Component>> components;
-
-    // Collect component's names to avoid duplicate components.
-    std::set<std::string> component_names;
-
     for (
         auto const& component_config :
         //! \ogs_file_param{prj__media__medium__phases__phase__components__component}
         config->getConfigSubtreeList("component"))
     {
         auto component = createComponent(component_config);
 
-        bool new_insertion = false;
-        std::tie(std::ignore, new_insertion) =
-            component_names.insert(component->name());
-        if (!new_insertion)
+        auto it = std::find_if(
+            components.begin(), components.end(), [&component](auto const& c) {
```

Comment:
Just a comment: one can also capture a new variable in the lambda, to avoid repeated evaluations.
```c++
auto f = [name = component->name()](auto const& c) { return c->name() == name; };
```
or if you desperately want a reference, then
```c++
... [name = std::cref(component->name())] ...
```

Reply:
Good point! I see.

Reply:
Changed.